### DISCUSSION TITLE
"Treat as 100% utilized" color

### DISCUSSION THREAD

**Discussion by alexandredpy** [netbox-community/netbox#7745] [Q&A] [answered]
Hello,
I hope this is the right place to ask this, I didn't want to create an issue for nothing...

Is it an expected feature, that when a subnet (prefix) is treated as 100% utilized, it's percentage color is grey ?

Theses 2 prefixes are currently treated as 100% utilized : 

![image](https://user-images.githubusercontent.com/18117508/140323531-78aaff83-7b04-4fe1-8247-2b2767afcc30.png)

I was hoping that it would appear red like this type of full subnet :
 
![image](https://user-images.githubusercontent.com/18117508/140323672-b159634a-f4c0-4a47-92f7-f3a3edd36f8f.png)

Thansk a lot,

**Comment by candlerb** [comment]
Can you show how to reproduce this with new prefixes?  What's different between the two examples you've given?

**Reply by alexandredpy** [reply]
Hello,
The red example is a /30 IPv4 subnet. The 2 hosts availables addresses are assigned, so the /30 subnet is full.

The grey example is a /126 IPv6 subnet. I only use 2 IPv6 adresses out of the 4 availables. I ticked the box "Treat as 100% utilized", so it appears as 100% but still in grey.

If you think it is an issue, tell me so that I will open an issue in the correct place.

**Reply by alexandredpy** [reply]
To reproduce, add a new prefix : (in my case a /126 IPv6 prefix):

![image](https://user-images.githubusercontent.com/18117508/140373683-0449f279-24c1-4ce7-a2ed-c4a8beb0707b.png)

Submit, (no addresses has been added yet), and it is shown as grey : 

![image](https://user-images.githubusercontent.com/18117508/140373897-8a487e0e-e1b5-4308-9075-b4fe34170cb5.png)

**Comment by ktims** [comment, marked as answer]
This behaviour is consistent with IPv4 when 'Mark utilized' is selected. Regardless of the actual utilization of the prefix, the 'progress bar' shows in grey. This might be intentional to indicate that it isn't 'live data'?

**Reply by alexandredpy** [reply]
This is my question :)

**Reply by jeremystretch** [reply]
Yep, this is intentional. The thresholds are disabled for the utilization bar because it's been set manually. You can find the code [here](https://github.com/netbox-community/netbox/blob/e04402ed570ad702f8f7f32ba5e2ecc3d51829f1/netbox/ipam/tables/ip.py#L166).

**Reply by alexandredpy** [reply]
well thanks for your answer jeremy!
